### PR TITLE
Update some forms to use new fields API

### PR DESCRIPTION
**Pull request by hypeJunction**
Cherry picks #10424
Also updates user settings form
Fixes an issue with field variables leaking to the input markup

### CODE REVIEW

File: engine/lib/output.php [hunk 1]
```diff
@@ -142,6 +142,11 @@ function elgg_format_attributes(array $attrs = array()) {
 	$attributes = [];
 
 	foreach ($attrs as $attr => $val) {
+		if (!is_string($attr) || $attr[0] == '#') {
```
Comment (jdalsem) [inline engine/lib/output.php]: why not fix the leaking?

Comment (hypeJunction) [inline engine/lib/output.php]: What do you mean? This fixes it.

Comment (jdalsem) [inline engine/lib/output.php]: you could wonder why the hashed attribute is even added.... this is fixing symptoms.. if you want to do that, we need a better check for invalid attribute_names (not only starting with a hash)

Comment (jdalsem) [inline engine/lib/output.php]: the leaking of hashed attributes was fixed by me in https://github.com/Elgg/Elgg/pull/10421/commits/fd9958c0f5ea824e99fc1881a3c416fb3342cdcb

Comment (jdalsem) [inline engine/lib/output.php]: > You want the input views to know what the parent field variables are. That's handy for view var filtering.

really? what is the use case? only parent field variables are #type, #help, #label... why would you need those?

Comment (hypeJunction) [inline engine/lib/output.php]: There will be others.

Comment (mrclay) [inline engine/lib/output.php]: If we're going to push arbitrary keys into input views (I get the use case), we should consolidate them as `#field_vars` or something and document them in the input views.

Comment (hypeJunction) [inline engine/lib/output.php]: I think I am going to change my position here. Let's filter out field vars until we have a compelling use case to pass them on

**Comment by hypeJunction** [comment]
Seems to have been fixed elsewhere

**Pull request closed without merging**